website/server/controllers/api-v3/tasks/groups.js

```diff
@@ -0,0 +1,183 @@
+ */
+api.assignTask = {
+  method: 'POST',
+  url: '/tasks/:taskId/assign/:assignedUserId',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    req.checkParams('taskId', res.t('taskIdRequired')).notEmpty().isUUID();
+    req.checkParams('assignedUserId', res.t('userIdRequired')).notEmpty().isUUID();
+
+    let reqValidationErrors = req.validationErrors();
+    if (reqValidationErrors) throw reqValidationErrors;
+
+    let user = res.locals.user;
+    let assignedUserId = req.params.assignedUserId;
+    let assignedUser = await User.findById(assignedUserId);
+
+    let taskId = req.params.taskId;
+    let task = await Tasks.Task.findByIdOrAlias(taskId, user._id);
+
+    if (!task) {
+      throw new NotFound(res.t('taskNotFound'));
+    }
+
+    if (!task.group.id) {
+      throw new NotAuthorized(res.t('onlyGroupTasksCanBeAssigned'));
+    }
+
+    let group = await Group.getGroup({user, groupId: req.params.groupId, populateLeader: false, fields: '_id'});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    if (group.leader !== user._id) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
+
+    await group.syncTask(task, assignedUser);
+
+    res.respond(201, task);
+  },
+};
+
+/**
+ * @api {post} /api/v3/tasks/:taskId/unassign/:assignedUserId Unassign a user from a task
+ * @apiDescription Unassigns a user to from a group task
+ * @apiVersion 3.0.0
+ * @apiName UnassignTask
+ * @apiGroup Task
+ *
+ * @apiParam {UUID} taskId The id of the task that will be assigned
+ * @apiParam {UUID} userId The id of the user that will be assigned to the task
+ *
+ * @apiSuccess data An object if a single task was created, otherwise an array of tasks
+ */
+api.unassignTask = {
+  method: 'POST',
+  url: '/tasks/:taskId/unassign/:assignedUserId',
+  middlewares: [authWithHeaders()],
+  async handler (req, res) {
+    req.checkParams('taskId', res.t('taskIdRequired')).notEmpty().isUUID();
+    req.checkParams('assignedUserId', res.t('userIdRequired')).notEmpty().isUUID();
+
+    let reqValidationErrors = req.validationErrors();
+    if (reqValidationErrors) throw reqValidationErrors;
+
+    let user = res.locals.user;
+    let assignedUserId = req.params.assignedUserId;
+    let assignedUser = await User.findById(assignedUserId);
+
+    let taskId = req.params.taskId;
+    let task = await Tasks.Task.findByIdOrAlias(taskId, user._id);
+
+    if (!task) {
+      throw new NotFound(res.t('taskNotFound'));
+    }
+
+    if (!task.group.id) {
+      throw new NotAuthorized(res.t('onlyGroupTasksCanBeAssigned'));
+    }
+
+    let group = await Group.getGroup({user, groupId: req.params.groupId, populateLeader: false, fields: '_id'});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    if (group.leader !== user._id) throw new NotAuthorized(res.t('onlyGroupLeaderCanEditTasks'));
+
+    await group.unlinkTask(task, assignedUser);
+
+    res.respond(201, task);
```

Comment:
Why is this 201 then: https://github.com/HabitRPG/habitrpg/blob/develop/website/server/controllers/api-v3/tasks.js#L92